service/src/main/java/uk/gov/hmcts/reform/fpl/controllers/PlacementController.java

```diff
@@ -111,4 +124,14 @@ private void removeTemporaryFields(CaseDetails caseDetails, String... fields) {
             caseDetails.getData().remove(field);
         }
     }
+
+    private void publishPlacementApplicationUploadEvent(CallbackRequest callbackRequest) {
+        applicationEventPublisher.publishEvent(
+            new PlacementApplicationEvent(callbackRequest, requestData.authorisation(), requestData.userId()));
+    }
+
+    private boolean isUpdatingExistingPlacementFilename(Placement previousPlacement, Placement newPlacement) {
+        return isNotEmpty(previousPlacement) && newPlacement.application.getFilename()
+            .equals(previousPlacement.application.getFilename());
```

Comment:
just curious, do we need to verify the contents of the file also?

Reply:
We may in the future need to verify the document's contents but not likely to happen in FPLA-821. Document reference would be unreliable as DM will regenerate new reference ID each file upload.